service/src/main/java/bio/terra/landingzone/stairway/flight/create/resource/step/BaseResourceCreateStep.java

```diff
@@ -64,23 +68,27 @@ public StepResult doStep(FlightContext context) throws InterruptedException, Ret
     var landingZoneId =
         getParameterOrThrow(
             context.getInputParameters(), LandingZoneFlightMapKeys.LANDING_ZONE_ID, UUID.class);
+    var azureLandingZoneRequest =
+        getParameterOrThrow(
+            context.getInputParameters(),
+            LandingZoneFlightMapKeys.LANDING_ZONE_CREATE_PARAMS,
+            LandingZoneRequest.class);
     try {
+      var stepDuration =
+          MetricUtils.configureTimerForLzStepDuration(
+              azureLandingZoneRequest.definition(), getResourceType());
+      var start = Instant.now().toEpochMilli();
       createResource(context, armManagers);
+      var finish = Instant.now().toEpochMilli();
+      stepDuration.record(Duration.ofMillis(finish - start));
     } catch (ManagementException e) {
-      var handled = maybeHandleManagementException(e);
-      if (handled.isPresent()) {
-        return handled.get();
-      }
-
-      if (StringUtils.equalsIgnoreCase(e.getValue().getCode(), "conflict")) {
-        logger.info(
-            RESOURCE_ALREADY_EXISTS, getResourceType(), billingProfile.getManagedResourceGroupId());
-        return StepResult.getStepResultSuccess();
-      }
-      logger.error(
-          FAILED_TO_CREATE_RESOURCE, getResourceType(), landingZoneId.toString(), e.toString());
-      return new StepResult(StepStatus.STEP_RESULT_FAILURE_FATAL, e);
+      return handleManagementException(
+          e,
+          landingZoneId.toString(),
+          azureLandingZoneRequest.definition(),
+          billingProfile.getManagedResourceGroupId());
     } catch (RuntimeException e) {
+      MetricUtils.incrementLandingZoneCreationFailure(azureLandingZoneRequest.definition());
```

Comment:
Ah ok, this is incrementing a counter to indicate the failure of a _step_ during the creation process, not the failure of teh creation process in it's entirety, right? The metric should be named (or namespaced) to indicate that.

Reply:
Yes. And in this case step's failure leads to the whole flight failure.